text/0006-equivocation-catastrophes.md

```diff
@@ -0,0 +1,111 @@
+2. Restarting the network with the new starting point.
+
+We will go into more detail about these steps below.
+
+### Social consensus
+
+For the initial version of this CEP, we assume that we have some kind of platform that enables open communication between validators and CasperLabs. The chosen platform should allow for storing, linking, referencing and displaying the whole public process and communication. We also should make sure that the content is in a format that can be displayed elsewhere.
+
+The first essential step will be determining the starting point for the network after the restart. The suggested approach here would be to agree upon the new initial global state, new set of validators and restart the network as if it was starting from scratch.
+
+A desirable goal for the initial global state would be to revert as few transactions from all the forks as possible, in order to limit the damage. Every transaction that was included in a block which appeared to be finalized for a time could have caused real world consequences (like, for example, a merchant sending ordered goods). Reverting any such transaction might potentially cause a loss to a user.
+
+In order to minimize losses, the following approach is proposed: first, choose the last finalized block as the starting point. If the equivocators caused multiple forks to appear, choose the tip of one of the forks. The exact criteria of choice will be left to the community to decide.
+
+After the block is chosen, take its global state and all transactions from the blocks in other forks and attempt to replay them on top of that global state. Since the order could matter, the community would have to agree on one. Call the resulting global state the "pre-initial state".
+
+As the last step, agree on the initial set of validators for the restarted network. The equivocators should be slashed, but there may be reasons to also exclude other validators, or even to start with a set of validators that has nothing in common with the validators from the stopped network. Once agreement has been reached, apply necessary changes to the pre-initial state: modify the validator stakes and change the entries in the auction contract. After this is applied, the resulting state will be the state used for the restart.
+
+After the users collectively decide upon the new initial state and the new initial set of validators, the network can be restarted.
+
+### Restarting the network
+
+The process of restarting the network would be very similar to the process of starting it. In its essence, it is just like starting the network from scratch, but with a few differences:
+
+- There is a different set of genesis validators - the new validators chosen by the community.
+- There is a different initial global state: it is not empty anymore, but it is the initial state prepared before.
+- Other details like the protocol version can differ as well, but they are not of the main concern for the restart process itself.
```

Comment:
…and we'd not start from height 0; in other words, we'd keep the linear chain from the original network.